### PR TITLE
init Reproducible Builds

### PR DESCRIPTION
**Pull request by hboutemy**
Signed-off-by: Hervé Boutemy <hboutemy@apache.org>

equivalent to https://github.com/antlr/antlr3/pull/210 that I did for ANTLR3

given ANTLR4 uses ANTLR3, to get a full reproducible build for ANTLR4, ANTLR3 version will require to be upgraded once https://github.com/antlr/antlr3/pull/209 has been merged and a release done

### CODE REVIEW

**Comment by parrt** [comment]
This seems harmless enough.

File: pom.xml [hunk 1]
```diff
@@ -163,6 +164,16 @@
 					<groupId>org.apache.maven.plugins</groupId>
 					<artifactId>maven-javadoc-plugin</artifactId>
 				</plugin>
+				<plugin>
```
Comment (parrt) [inline pom.xml:167]: Why do we need the extra dependencies here? Does the outputTimestamp thing need these?

Comment (hboutemy) [reply pom.xml:167]: @parrt
it's not really new dependencies: it's an update of plugin versions for plugins that are already used (default Maven lifecycle for jars), to use plugins that support Reproducible Builds (use the outputTimestamp)

**Comment by hboutemy** [comment]
@parrt I added a second commit to workaround antlr3 non-reproducible output, as discussed in https://github.com/antlr/antlr3/pull/209
now, the build for ANTLR4 is fully reproducible on my side, as tested with:
```
mvn clean install -DskipTests && mvn clean package artifact:compare -DskipTests
```

**Pull request merged**